app/core/views/submission.py

```diff
@@ -0,0 +1,109 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.http import HttpResponseBadRequest
+from django.shortcuts import redirect, render
+from django.urls import reverse_lazy
+from django.views.generic.edit import DeleteView
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+
+from ..forms import SubmissionForm, ContainerForm
+from ..models import Submission
+
+
+class OwnerMatchMixin(LoginRequiredMixin, UserPassesTestMixin):
```

Comment:
What does this class do?

Reply:
This mixin on a view will make sure the user accessing the object in the view is the owner of that object. test_func is the Django builtin that will be used in the standard Django object-based views